src/components/DistanceRequest/index.js

```diff
@@ -183,13 +211,27 @@ function DistanceRequest({transactionID, report, transaction, route, isEditingRe
                 />
             </View>
             <View style={[styles.w100, styles.pt2]}>
+                {/* show error message if there is route error or there are less than 2 routes and user has tried submitting */}
+                {((userTriedSubmitting && _.size(validatedWaypoints) < 2) || hasRouteError) && (
+                    <DotIndicatorMessage
+                        style={[styles.mh4, styles.mv3]}
+                        messages={getError()}
+                        type="error"
+                    />
+                )}
                 <Button
                     success
                     allowBubble
                     pressOnEnter
                     style={[styles.w100, styles.mb4, styles.ph4, styles.flexShrink0]}
-                    onPress={() => onSubmit(waypoints)}
-                    isDisabled={_.size(validatedWaypoints) < 2 || (!isOffline && (hasRouteError || isLoadingRoute || isLoading))}
+                    onPress={() => {
+                        setUserTriedSubmitting(true);
+                        // if there is any error or loading state, don't let user go to next page
+                        if (_.size(validatedWaypoints) < 2 || hasRouteError || isLoadingRoute || isLoading) {
+                            return;
+                        }
```

Comment:
Could you please move this logic inside of `onSubmit`?

Reply:
Ah, I didn't realize that `onSubmit()` was from props. OK, then the proper thing to do is to move this to a function that is defined above the `return`, add a `useCallback()` and pass the function here as a reference.